src/text_shaper/directwrite_shaper.cpp

```diff
@@ -66,6 +66,60 @@ namespace
         }
         return font_slant::normal;
     }
+
+    void renderGlyphRunToBitmap(IDWriteGlyphRunAnalysis* _glyphAnalysis, RECT& _textureBounds, bitmap_format _targetFormat, std::vector<uint8_t>::iterator& _it)
+    {
+        const auto width = _textureBounds.right - _textureBounds.left;
+        const auto height = _textureBounds.bottom - _textureBounds.top;
+
+        std::vector<uint8_t> tmp;
+        tmp.resize(height *width * 3);
+
+        auto hr = _glyphAnalysis->CreateAlphaTexture(
+            DWRITE_TEXTURE_CLEARTYPE_3x1,
+            &_textureBounds,
+            tmp.data(),
+            tmp.size()
+        );
+
+        for (auto i = 0; i < height; i++)
```

Comment:
Would you want to give `range::views::iota(0, N)` a try? Ah example use would be here: 
https://github.com/contour-terminal/contour/blob/abdae0192006275b31a6e2e5222755e5b1176924/src/terminal/Grid.cpp#L557-L258

It avoid the need of having the iterator variable being noon-const (mutable) even though it is only used read-only in the loop body.

Reply:
I am too dumb to actually make it work 😅 . The compile hates me. I'll give it a try in future PRs.